File: 1260.java
package boj;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

public class boj1260 {
	static boolean check_dfs[];
	static boolean check_bfs[];
	static boolean list[][];
	static Queue<Integer> q;
	private static void dfs(int n) {
		check_dfs[n] = true;
		System.out.print(n + " ");
		for(int i = 0;i<list[n].length;i++){
			if(list[n][i] == true && check_dfs[i] == false){
				dfs(i);
			}
		}
	}
	
	private static void bfs(int n){
		if(q.isEmpty()){
			q.add(n);
		}
		check_bfs[n] = true;
		
		while(!q.isEmpty()){
			int v = q.peek();
			System.out.print(v+ " ");
			for(int i=0;i<list[v].length;i++){
				if(list[v][i]==true && check_bfs[i]==false){
					q.add(i);
					check_bfs[i] = true;
				}
			}
			q.remove();
		}
	}
	
	public static void main(String[] args) {
		Scanner s = new Scanner(System.in);
		int n = s.nextInt();
		int m = s.nextInt();
		int start = s.nextInt();
		
		check_dfs = new boolean[n+1];
		check_bfs = new boolean[n+1];
		list = new boolean[n+1][n+1];
		q = new LinkedList();
		
		for(int i=0;i<m;i++){
			int u = s.nextInt();
			int v = s.nextInt();
			list[u][v] = list[v][u] = true;
		}
		dfs(start);
		System.out.println();
		bfs(start);
	}
}

File: 1260.cpp
/* 
  자바로 구현한 코드 다시 C++로 작성
*/
#include <iostream>
#include <string.h>
#include <queue>
using namespace std;
bool check[1001] = { false };
bool graph[1001][1001] = { false } ;
//인접행렬
void dfs(int n) {
	check[n] = true;
	cout << n << " ";
	for (int i = 1; i < sizeof(graph[n]); i++) {
		if (graph[n][i] && check[i] == false)
			dfs(i);
	}
}

void bfs(int n) {
	memset(check, false, sizeof(check));
	queue<int> q;
	q.push(n);
	check[n] = true;
	while (!q.empty()) {
		int v = q.front();
		cout << v << " ";
		for (int i = 0; i < sizeof(graph[v]); i++) {
			if (graph[v][i] && check[i] == false) {
				q.push(i);
				check[i] = true; //중요!
			}
		}
		q.pop();
	}
}

int main() {
	int n, m, start;
	cin >> n >> m >> start;
	//인접행렬
	for (int i = 0; i < m; i++) {
		int v,e;
		cin >> v >> e;
		graph[v][e] = graph[e][v] = true;
	}
	dfs(start);
	cout << endl;
	bfs(start);
	cout << endl;
}
